include/nbl/asset/ICPUPipeline.h

```diff
@@ -121,31 +119,28 @@ class ICPUPipeline : public IAsset, public PipelineNonAssetBase, public ICPUPipe
 
         inline core::smart_refctd_ptr<IAsset> clone(uint32_t _depth = ~0u) const override final
         {
+            if (!getLayout()) return nullptr;
```

Comment:
you can just assert it, because you can't make a pipelien without a non-null layout